awx/main/tasks.py

```diff
@@ -466,6 +466,8 @@ def cluster_node_heartbeat():
         if inst.hostname == settings.CLUSTER_HOST_ID:
             this_inst = inst
             instance_list.remove(inst)
+        elif inst.version.startswith('ansible-runner'):  # TODO: use proper field when introduced
+            continue
```

Comment:
Jobs won't really work in the main branch, and this is the necessary fix to resolve that.

Execution nodes won't update capacity on a regular basis whatsoever. This set capacity of other nodes to 0 if they their heartbeat did not work. Execution nodes have no heartbeat, so that's the reason for excluding them here.

Reply:
If they don’t have a heartbeat, how do we know that they are still healthy?
Also, won’t they use up capacity as they execute jobs?